zebra/zebra_rib.c

```diff
@@ -1075,7 +1075,8 @@ static void rib_process(struct route_node *rn)
 		}
 
 		/* Infinite distance. */
-		if (re->distance == DISTANCE_INFINITY) {
+		if (re->distance == DISTANCE_INFINITY &&
+		    re->type != ZEBRA_ROUTE_KERNEL) {
```

Comment:
We are keeping the kernel route with distance 255 as installed but this will not be redistributed to any other protocols due to below check in function zebra_redistribute. 
			if (newre->distance == DISTANCE_INFINITY)
				continue;

I assume that is expected and intention is only to deal this route at zera level?

Reply:
yes